inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/manager/MessageReadinessManager.kt

```diff
@@ -56,8 +56,9 @@ internal interface MessageReadinessManager {
     private class MessageReadinessManagerImpl : MessageReadinessManager {
         @WorkerThread
         @Suppress("LongMethod", "ReturnCount")
-        override fun getNextDisplayMessage(): Message? {
-            val messageList: List<Message> = ReadyForDisplayMessageRepository.instance().getAllMessagesCopy()
+        override fun getNextDisplayMessage(ignored: List<Message>): Message? {
+            var messageList: List<Message> = ReadyForDisplayMessageRepository.instance().getAllMessagesCopy()
+            messageList.subtract(ignored)
```

Comment:
I'm not sure which is faster, subtracting the whole list at the start or just use the initial list and check if the message is part of the ignore list inside the `for` loop.

```suggestion
            for (message in messageList) {
               message in ignored
               ...
            }
```

What do you think?

Reply:
Yeah I was thinking about that. I'm not sure how complex `subtract()` is. Traversing `ignored` list for each message in `messageList` doesn't sound like a faster solution but considering that `ignored` list will contain just a few elements I think there is no much difference here. I'd choose the one that is easier to understand in the code. Subtract is more elegant to me but your solution looks more clear. I'm not sure. What is your vote?

Reply:
either is ok with me.
we can retain the `subtract` for now and just update it when we encounter issue